lib/stdlib/src/gen.erl

```diff
@@ -198,17 +199,41 @@ call(Process, Label, Request, Timeout)
     Fun = fun(Pid) -> do_call(Pid, Label, Request, Timeout) end,
     do_for_proc(Process, Fun).
 
-do_call(Process, Label, Request, Timeout) when is_atom(Process) =:= false ->
+-dialyzer({no_improper_lists, do_call/4}).
+
+do_call(Process, Label, Request, infinity)
+  when (is_pid(Process)
+        andalso (node(Process) == node()))
+       orelse (element(2, Process) == node()
+               andalso is_atom(element(1, Process))
+               andalso (tuple_size(Process) =:= 2)) ->
     Mref = erlang:monitor(process, Process),
+    %% Local without timeout; no need to use alias since we unconditionally
+    %% will wait for either a reply or a down message which corresponds to
+    %% the process being terminated (as opposed to 'noconnection')...
+    Process ! {Label, {self(), Mref}, Request},
+    receive
+        {Mref, Reply} ->
+            erlang:demonitor(Mref, [flush]),
+            {ok, Reply};
+        {'DOWN', Mref, _, _, Reason} ->
+            exit(Reason)
+    end;
+do_call(Process, Label, Request, Timeout) when is_atom(Process) =:= false ->
+    Mref = erlang:monitor(process, Process, [{alias,demonitor}]),
+
+    Tag = [alias | Mref],
```

Comment:
Is the compiler capable of tracking selective receives when the reference is stored in a variable that is then used in the `receive`? Or does the compiler need to be improved here? Also, do we need to teach the compiler about `monitor/3` when talking about selective receives?

Reply:
I forgot to make the compiler aware of `monitor/3`.

Reply:
Just pushed a fix of this